bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/handler/It600Handler.java

```diff
@@ -0,0 +1,265 @@
+import static org.openhab.binding.salus.internal.SalusBindingConstants.It600Device.HoldType.TEMPORARY_MANUAL;
+import static org.openhab.binding.salus.internal.SalusBindingConstants.SalusDevice.DSN;
+import static org.openhab.core.thing.ThingStatus.OFFLINE;
+import static org.openhab.core.thing.ThingStatus.ONLINE;
+import static org.openhab.core.thing.ThingStatusDetail.BRIDGE_UNINITIALIZED;
+import static org.openhab.core.thing.ThingStatusDetail.COMMUNICATION_ERROR;
+import static org.openhab.core.thing.ThingStatusDetail.CONFIGURATION_ERROR;
+
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.util.ArrayList;
+import java.util.Optional;
+import java.util.Set;
+import java.util.SortedSet;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.salus.internal.rest.DeviceProperty;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.BridgeHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Martin Grześlowski - Initial contribution
+ */
+@NonNullByDefault
+public class It600Handler extends BaseThingHandler {
+    private static final BigDecimal ONE_HUNDRED = new BigDecimal(100);
+    private static final Set<String> REQUIRED_CHANNELS = Set.of("ep_9:sIT600TH:LocalTemperature_x100",
+            "ep_9:sIT600TH:HeatingSetpoint_x100", "ep_9:sIT600TH:SetHeatingSetpoint_x100", "ep_9:sIT600TH:HoldType",
+            "ep_9:sIT600TH:SetHoldType");
+    private final Logger logger;
+    @NonNullByDefault({})
+    private String dsn;
+    @NonNullByDefault({})
+    private CloudApi cloudApi;
+
+    public It600Handler(Thing thing) {
+        super(thing);
+        logger = LoggerFactory.getLogger(It600Handler.class.getName() + "[" + thing.getUID().getId() + "]");
+    }
+
+    @Override
+    public void initialize() {
+        try {
+            logger.debug("Initializing thing...");
+            internalInitialize();
+        } catch (Exception e) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR,
+                    "Error occurred while initializing Salus device! " + e.getLocalizedMessage());
+        }
+    }
+
+    private void internalInitialize() {
+        {
+            var bridge = getBridge();
+            if (bridge == null) {
+                updateStatus(OFFLINE, BRIDGE_UNINITIALIZED,
+                        "There is no bridge for this thing. Remove it and add it again.");
+                return;
+            }
+            var bridgeHandler = bridge.getHandler();
+            if (!(bridgeHandler instanceof CloudBridgeHandler cloudHandler)) {
```

Comment:
```suggestion
            if (!(bridge.getHandler() instanceof CloudBridgeHandler cloudHandler)) {
```

Noty sure if this is still the right way to get the bridge, this has changed recently might have to re-visit this later on.

Reply:
I'm not getting any deprecated warning so I assume it's fine